render/render.go

```diff
@@ -83,6 +87,30 @@ func mkWriter(path string) (io.Writer, func(), error) {
 
 /* Template functions */
 
+var regexCanonical = regexp.MustCompile(`^(?P<version>v[0-9.]+)`)
+var regexRevision = regexp.MustCompile(`[^-]+-[^-]+-(?P<revision>[a-fA-Z0-9]+)`)
+
+func CanonicalVersion(in string) string {
```

Comment:
This could use a comment because at least for me "canonical version" doesn't have a specific meaning (though it looks like the definition is "same as semver canonical but with +incompatible if necessary" which still does not help much tbh)

Reply:
Added some examples to the godoc and unit tests